Source/WebCore/style/StyleBuilderConverter.h

```diff
@@ -1891,15 +1896,16 @@ inline float BuilderConverter::convertOpacity(BuilderState& builderState, const
     return std::max(0.0f, std::min(1.0f, opacity));
 }
 
-inline String BuilderConverter::convertSVGURIReference(BuilderState& builderState, const CSSValue& value)
+inline Style::URL BuilderConverter::convertSVGURIReference(BuilderState& builderState, const CSSValue& value)
 {
+    if (auto url = dynamicDowncast<CSSURLValue>(value))
+        return Style::toStyle(url->url(), builderState);
+
     auto* primitiveValue = requiredDowncast<CSSPrimitiveValue>(builderState, value);
     if (!primitiveValue)
-        return { };
+        return Style::URL::empty();
```

Comment:
If the new code is correct then we leave this paragraph out entirely:
```suggestion
```
This relies on knowing that the change from null to empty is good or at least harmless, and that there are no side effects of `requiredDowncast`.

Reply:
Unfortunately, there are side effects to `requiredDowncast`. It sets `setCurrentPropertyInvalidAtComputedValueTime` on failures.